frontend/src/pages/modelRegistry/ModelRegistryCoreLoader.tsx

```diff
@@ -32,8 +32,23 @@ const ModelRegistryCoreLoader: React.FC<ModelRegistryCoreLoaderProps> =
       modelRegistryServicesLoadError,
       modelRegistryServices,
       preferredModelRegistry,
+      updatePreferredModelRegistry,
     } = React.useContext(ModelRegistrySelectorContext);
 
+    const modelRegistryFromRoute = modelRegistryServices.find(
+      (mr) => mr.metadata.name === modelRegistry,
+    );
+
+    React.useEffect(() => {
+      if (
+        modelRegistryFromRoute &&
+        preferredModelRegistry?.metadata.name !== modelRegistryFromRoute.metadata.name
+      ) {
+        updatePreferredModelRegistry(modelRegistryFromRoute);
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I don't think we should disable the `exhaustive-deps` lint rule here, if you remove this line it just makes you add `preferredModelRegistry?.metadata.name` to the dependencies which works fine.